src/host/main.cpp

```diff
@@ -661,16 +697,22 @@ int main(int argc, char** argv)
 
     asynchost::RPCConnections rpc(writer_factory);
     rpc.register_message_handlers(bp.get_dispatcher());
-    rpc.listen(0, rpc_address.hostname, rpc_address.port);
-    if (!rpc_address_file.empty())
+
+    std::string rpc_addresses;
+    for (auto& interface : rpc_interfaces)
     {
-      files::dump(
-        fmt::format("{}\n{}", rpc_address.hostname, rpc_address.port),
-        rpc_address_file);
+      rpc.listen(0, interface.rpc_address.hostname, interface.rpc_address.port);
+      rpc_addresses += fmt::format(
```

Comment:
When we write the (possibly resolved) RPC addresses to file, we write all of them.